src/NUnitFramework/framework/Internal/Execution/TextMessageWriter.cs

```diff
@@ -252,7 +305,9 @@ private void WriteExpectedLine(object expected, Tolerance tolerance)
         {
             Write(Pfx_Expected);
             Write(MsgUtils.FormatValue(expected));
-
+            if (diffTypes) {
```

Comment:
This writes the type whenever the two types differ, but we only wanted to do it when the types differ and the display strings for the two values are identical.